test/test-tools.cpp

```diff
@@ -0,0 +1,65 @@
+#include <magic.h>
+#include "gtest/gtest.h"
+#include "../src/tools.h"
+
+
+// stub from zimwriterfs.cpp
+std::string directoryPath = ".";
+bool isVerbose() { return true; }
+bool inflateHtmlFlag = false;
+bool uniqueNamespace = false;
+magic_t magic;
+
+namespace
+{
+class ToolsTest : public ::testing::Test
+{
+public:
+  ToolsTest() {
+    magic = magic_open(MAGIC_MIME);
+    magic_load(magic, NULL);
+  }
+
+  ~ToolsTest() {
+      magic_close(magic);
+  }
+};
+
+TEST(tools, isOutofBounds)
+{
+    ASSERT_FALSE(isOutofBounds("", ""));
+    ASSERT_TRUE(isOutofBounds("../../..", ""));
+    ASSERT_TRUE(isOutofBounds("../", ""));
+}
+
+TEST(tools, normalize_link)
+{
+    ASSERT_EQ(normalize_link("/a", "/b"), "a");
+
+    // not absolute
+    ASSERT_EQ(normalize_link("a", "/b"), "/b/a");
+    ASSERT_EQ(normalize_link("../a", "/b/c"), "/b/a");
+    ASSERT_EQ(normalize_link(".././a", "/b/c"), "/b/a");
+    ASSERT_EQ(normalize_link("../a/b/aa#localanchor", "/b/c"), "/b/a/b/aa");
+    ASSERT_EQ(normalize_link("../a/b/aa?localanchor", "/b/c"), "/b/a/b/aa");
+}
+
+TEST(tools, addler32)
+{
+    ASSERT_EQ(adler32("sdfkhewruhwe8"), 640746832);
```

Comment:
Test with at least two different values (and a empty string).

`int adler32(std::string buf) { return 640746832; }` is a implementation passing this test.